fenix/app/src/main/java/org/mozilla/fenix/BrowsingModeBinding.kt

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix
+
+import android.view.Window
+import android.view.WindowManager
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.distinctUntilChangedBy
+import kotlinx.coroutines.withContext
+import mozilla.components.lib.state.helpers.AbstractBinding
+import org.mozilla.fenix.browser.browsingmode.BrowsingMode
+import org.mozilla.fenix.components.AppStore
+import org.mozilla.fenix.components.appstate.AppState
+import org.mozilla.fenix.theme.ThemeManager
+import org.mozilla.fenix.utils.Settings
+
+/**
+ * Binding to react to Private Browsing Mode changes in AppState.
+ *
+ * @param appStore AppStore to observe state changes from.
+ * @param themeManager Theme will be updated based on state changes.
+ * @param retrieveWindow Get window to update privacy flags for.
+ * @param settings Determine user settings for privacy features.
+ * @param ioDispatcher Dispatcher to launch disk reads. Exposed for test.
+ */
+class BrowsingModeBinding(
```

Comment:
I believe having this binding will allow us to remove the secure window flag handling in other fragments and just have a centralized place where we handle this. I am wondering if there is any plans to clean this up in other fragments afterrwards. Example can be seen in https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/settings/PrivateBrowsingFragment.kt#48-54

Reply:
At a glance it does look like this is at least worth a slightly more thorough look, so I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1872945